src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -702,18 +639,28 @@ async Task<InProgressState> CollapseInProgressStateAsync(InProgressState initial
                                 }
                                 else
                                 {
-                                    var metadataReference = await compilationState.GetMetadataReferenceAsync(
-                                        projectReference, this.ProjectState, cancellationToken).ConfigureAwait(false);
+                                    var metadataReference = isFrozen
+                                        ? compilationState.GetPartialMetadataReference(projectReference, this.ProjectState)
+                                        : await compilationState.GetMetadataReferenceAsync(projectReference, this.ProjectState, cancellationToken).ConfigureAwait(false);
```

Comment:
that big chunk that got ripped out was merged into here.  basically, now we can setup teh references in one uniform location.  we do have differing behavior if we're frozen or not, but that's desirable.

Reply:
note: i'm also considering updating GetMetadtareferenceAsync to take in the isFrozen flag.